src/mixins/form-text.js

```diff
@@ -100,38 +100,44 @@ export default {
     stringifyValue(value) {
       return isUndefined(value) || isNull(value) ? '' : String(value)
     },
-    getFormatted(value, event, force = false) {
+    getFormatted(value, evt, force = false) {
       value = this.stringifyValue(value)
       if ((!this.lazyFormatter || force) && isFunction(this.formatter)) {
-        value = this.formatter(value, event)
+        return this.formatter(value, evt)
       }
       return value
     },
     updateValue(value) {
       value = this.stringifyValue(value)
-      if (this.localValue !== value) {
-        // keep the input set to the value before modifiers
+      if (value !== this.localValue) {
+        // Keep the input set to the value before modifiers
         this.localValue = value
         if (this.number) {
-          // Emulate .number modifier behaviour
+          // Emulate `.number` modifier behaviour
           const num = parseFloat(value)
           value = isNaN(num) ? value : num
         } else if (this.trim) {
-          // Emulate .trim modifier behaviour
+          // Emulate `.trim` modifier behaviour
           value = value.trim()
         }
         // Update the v-model
         this.$emit('update', value)
+      } else if (value !== this.$refs.input.value) {
+        // When the `localValue` hasn't changed but the actual input value
+        // is out of sync, make sure to change it to the given one
+        this.$refs.input.value = value
```

Comment:
One thing to note about this is that it can cause the user's cursor/insert point to jump.  Although in general, formatters will have that effect as well.

Reply:
Yeah, I think we can/have to live with that.

Reply:
May want to add an `/*istanbul ignore next */` above the line, as it will be difficult to test this scenario